bitnami/discourse/templates/_helpers.tpl

```diff
@@ -243,18 +265,35 @@ Return the PostgreSQL User
 {{- end -}}
 
 {{/*
-Return the PostgreSQL Secret
+Return true if a secret object for Postgres should be created
+*/}}
+{{- define "discourse.databaseCreateSecret" -}}
+{{- if or (and .Values.postgresql.enabled .Values.postgresql.existingSecret) (and (not .Values.postgresql.enabled) .Values.externalDatabase.existingSecret) }}
```

Comment:
Nice! Here the restriction is complex so a helper for whether to create the secret or not comes handy!!

Reply:
Nevertheless, this statement is not OK. Let me exemplify where we should create a secret or not.
Let `pe` be `postgres.enabled`, `es` be `postgres.existingSecret ` and `ees` be `externalDatabase.existingSecret`:

|        | es | not es | ees | not ees |
|--------|----|--------|-----|---------|
| **pe**     | NO | NO     | NO  | NO      |
| **not pe** | NO | NO     | NO  | YES     |

If `postgres.enabled` => Either the subchart or the externalSecret will handle the credentials
If not `postgres.enabled` => You must create a secret only in the case that the external one is not provided.

So the restriction (correct me if I am wrong) should be:
```yaml
if and (not .Values.postgresql.enabled) (not .Values.externalDatabase.existingSecret)
```

Reply:
Correct, good catch..this become quite confusing since the name of the secret is managed somewhere else...apologize for the confusion